lib/request.js

```diff
@@ -161,7 +162,10 @@ Object.defineProperties(Request.prototype, {
   },
   headers: {
     get () {
-      return this.raw.headers
+      return this.processedHeaders || this.raw.headers
+    },
+    set (headers) {
+      this.processedHeaders = headers
```

Comment:
Would it be better to: 

```suggestion
      for (const [header, value] of Object.entries(headers)) {
        this.setHeader(header, value)
      }
```

I think this would obviate the need to define `processedHeaders` and require only adding the `set` method.

Reply:
Done with a little change since the `IncomingMessage` has only a `headers` getter